projects/ion/src/lib/link/link.component.html

```diff
@@ -0,0 +1,51 @@
+<span class="link-container" [class.link-container--disabled]="disabled">
+  @if (!link) {
+    <button
+      data-testid="ion-link"
+      class="ion-link"
+      [class.ion-link--left-icon]="iconSide === 'left'"
+      [class.ion-link--icon-only]="!label || label === ''"
+      [attr.disabled]="disabled"
+      (click)="onClick()">
+      @if (label && label.length) {
+        <span
+          data-testid="ion-link-label"
+          [class.ion-link__label--bold]="bold"
+          [ngClass]="'ion-link__label ion-link__label--' + size"
+          >{{ label }}
+        </span>
```

Comment:
I don't even know if this could be made on Angular 17, but I guess you could create a ng-template for this and for the icon section, since you repeating in both `button` and `a`.